controller/tests/controller_server/csi_controller_server_test.py

```diff
@@ -1264,6 +1260,151 @@ def test_controller_get_capabilities(self):
         self.servicer.ControllerGetCapabilities(request, context)
 
 
+class TestControllerServerExpandVolume(AbstractControllerTest):
+
+    def get_create_object_method(self):
+        return self.servicer.ControllerExpandVolume
+
+    def get_create_object_response_method(self):
+        return csi_pb2.ControllerExpandVolumeResponse
+
+    def setUp(self):
+        super().setUp()
+        self.fqdn = "fqdn"
+        self.mediator = XIVArrayMediator("user", "password", self.fqdn)
+        self.mediator.client = Mock()
+
+        self.storage_agent = MagicMock()
+        self.storage_agent.get_mediator.return_value.__enter__.return_value = self.mediator
+
+        self.servicer = ControllerServicer(self.fqdn)
+
+        self.access_types = csi_pb2.VolumeCapability.AccessMode
+        self.fs_type = "ext4"
+
+        self.request.volume_capability = csi_pb2.VolumeCapability(
+            access_mode=csi_pb2.VolumeCapability.AccessMode(mode=self.access_types.SINGLE_NODE_WRITER),
+            mount=csi_pb2.VolumeCapability.MountVolume(fs_type=self.fs_type))
+        self.request.parameters = {}
+        self.capacity_bytes = 10
+        self.request.capacity_range = Mock()
+        self.request.capacity_range.required_bytes = self.capacity_bytes
+        self.volume_id = "vol-id"
+        self.request.volume_id = "{}:{}".format("xiv", self.volume_id)
+        self.request.volume_content_source = None
+        self.context = utils.FakeContext()
+
+    def _prepare_expand_volume_mocks(self, storage_agent):
+        storage_agent.return_value = self.storage_agent
+
+        self.mediator.expand_volume = Mock()
+        self.mediator.get_object_by_id = Mock()
+        self.mediator.get_object_by_id.return_value = utils.get_mock_mediator_response_volume(self.capacity_bytes,
+                                                                                              volume_name,
+                                                                                              self.volume_id,
+                                                                                              "a9k")
+
+    @patch("controller.controller_server.csi_controller_server.get_agent")
+    def test_expand_volume_with_zero_size(self, storage_agent):
+        self._prepare_expand_volume_mocks(storage_agent)
+
+        self.request.capacity_range.required_bytes = 0
+        self.servicer.ControllerExpandVolume(self.request, self.context)
+        self.assertEqual(self.context.code, grpc.StatusCode.OK)
+        self.mediator.expand_volume.assert_called_once_with(volume_id=self.volume_id, required_bytes=0)
+
+    @patch("controller.controller_server.csi_controller_server.get_agent")
+    def test_expand_volume_with_size_too_large_fail(self, storage_agent):

```

Comment:
what about size too small?

Reply:
we have a default for it

Reply:
where? and how does it justify not unit testing such scenario?

Reply:
1. k8s wont let you expand less then current size
1 its in [CreateVolume](https://github.com/IBM/ibm-block-csi-driver/blob/ca8cf8c3f7eb8caec6a0337bda44941c13982549/controller/controller_server/csi_controller_server.py#L97)

Reply:
we shouldn't rely on current k8s implementation details - the spec allows k8s to change that behavior, and it shouldn't break our code

Reply:
added validation and UT